thinc/config.py

```diff
@@ -456,6 +459,20 @@ def from_disk(
         return self.from_str(text, interpolate=interpolate, overrides=overrides)
 
 
+def mask_positional_args(name: str) -> List[Optional[str]]:
+    """Create a section name representation that masks names
+    of positional arguments to retain their order in sorts."""
+
+    stable_name = cast(List[Optional[str]], name.split("."))
+
+    # Remove names of sections that are a positional arugment.
+    for i in range(1, len(stable_name)):
+        if stable_name[i - 1] == "*":
+            stable_name[i] = None
```

Comment:
I guess you prefer using `None` here to avoid any (however unlikely) clashes with names of other arguments? It would make the typing a little cleaner to have only strings, and I get a little nervous running the `None` values through `sort`, but I guess it's really just fine.